### PR TITLE
Redo collect active elem

### PR DESCRIPTION
**Pull request by drwells**
Followup to #826: I replaced the recursive algorithm for computing which elements lie on a given patch with one based on bounding boxes. The bounding boxes are computed from the locations of the displaced nodes and quadrature points of each element.

This is a bit more conservative than the old approach since we use bounding boxes instead of excluding elements that do not currently have any quadrature points on the current element.

### CODE REVIEW

File: ibtk/src/utilities/libmesh_utilities.cpp [hunk 1]
```diff
@@ -301,7 +301,6 @@ std::vector<libMeshWrappers::BoundingBox>
 get_local_active_element_bounding_boxes(const libMesh::MeshBase& mesh, const libMesh::System& X_system)
 {
     const unsigned int dim = mesh.mesh_dimension();
```
Comment (drwells) [inline ibtk/src/utilities/libmesh_utilities.cpp]: @AminKolah here's at least one bug with the surface mesh code: `mesh_dimension()` is the topological dimension but for bounding boxes we want the spatial dimension. I'll fix this.

**Comment by drwells** [comment]
I have no idea how the ex5 test managed to ever work when things were this broken: I'll check that next.

Either way I fixed a bunch of bugs regarding the partitioning of surface meshes: things should now work correctly. I added a bunch of new tests and also checked things visually. Here is a nice picture of our IB partitioning for a circle

visit0000

and another in 3D just for fun :)
visit0001

**Comment by drwells** [comment]
We now have over 200 tests :tada:

File: ibtk/src/utilities/libmesh_utilities.cpp [hunk 2]
```diff
@@ -300,8 +301,8 @@ get_local_active_element_bounding_boxes(const libMesh::MeshBase& mesh,
 std::vector<libMeshWrappers::BoundingBox>
 get_local_active_element_bounding_boxes(const libMesh::MeshBase& mesh, const libMesh::System& X_system)
 {
-    const unsigned int dim = mesh.mesh_dimension();
-    TBOX_ASSERT(dim <= LIBMESH_DIM);
+    const unsigned int spacedim = mesh.spatial_dimension();
```
Comment (boyceg) [inline ibtk/src/utilities/libmesh_utilities.cpp]: `mesh_dim` ?

Comment (drwells) [reply ibtk/src/utilities/libmesh_utilities.cpp]: I'm not sure I understand the question: what do you have in mind?

Comment (boyceg) [reply ibtk/src/utilities/libmesh_utilities.cpp]: The way that I think of it, at least, is that `NDIM` is the "spatial dimension". So I am suggesting:
```suggestion
    const unsigned int mesh_dim = mesh.spatial_dimension();
```

Comment (drwells) [reply ibtk/src/utilities/libmesh_utilities.cpp]: I agree with the first part but not the second. `mesh.mesh_dimension()` is the topological dimension of the mesh (which is not what we want): `mesh_dim` looks too much like that so I would prefer to write `spacedim` or `space_dimension`

**Comment by AminKolah** [comment]
> I have no idea how the ex5 test managed to ever work when things were this broken: I'll check that next.
> 
> Either way I fixed a bunch of bugs regarding the partitioning of surface meshes: things should now work correctly. I added a bunch of new tests and also checked things visually. Here is a nice picture of our IB partitioning for a circle
> 
> visit0000
> 
> and another in 3D just for fun :)
> visit0001

Nice! ..I never checked the codim-0 version of the ex5 with IBFEMethod (at least not recently), but the one with surface mesh was definitely working as expected on my branch...I did realize that in some parts of the libmesh_utilities.h it was assumed that NDIM the mesh dimensions for both solid and the fluid mesh (which would be only true for volumetric mesh)

**Comment by drwells** [comment]
@boyceg approved so I will merge so that @AminKolah can get on with his life :)

**Pull request merged**